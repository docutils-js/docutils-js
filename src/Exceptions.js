export class UnimplementedError extends Error {
    constructor(message, ...params) {
	super(...params);
	if(Error.captureStackTrace) {
	    Error.captureStackTrace(this, UnimplementedError);
	}
	this.message = message;
    }
}


export class EOFError extends Error {
    constructor(...params) {
	super(...params);
	if(Error.captureStackTrace) {
	    Error.captureStackTrace(this, EOFError);
	}
    }
}


export class InvalidArgumentsError extends Error {
    constructor(message, ...params) {
	super(...params);
	this.message = message;
	if(Error.captureStackTrace) {
	    Error.captureStackTrace(this, InvalidArgumentsError);
	}
    }
}
    
export const InvalidArgumentError = InvalidArgumentsError
export class SystemMessage extends Error {
    constructor(msg, level, ...params) {
	super(...params);
	this.message = msg.astext();
	this.msg = msg;
	this.level = level;
	if(Error.captureStackTrace) {
	    Error.captureStackTrace(this, SystemMessage);
	}
    }
}
    
