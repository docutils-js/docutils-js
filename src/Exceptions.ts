import {INode} from "./types";

export class UnimplementedError extends Error {
    // @ts-ignore
    public constructor(message?: string, ...params) {
        super(...params);
        /* instanbul ignore else */
        if (Error.captureStackTrace) {
            Error.captureStackTrace(this, UnimplementedError);
        }
        this.message = message || '';
    }
}


export class EOFError extends Error {
    public constructor(...params: any[]) {
        super(...params);
        /* instanbul ignore else */
        if (Error.captureStackTrace) {
            Error.captureStackTrace(this, EOFError);
        }
    }
}


export class InvalidArgumentsError extends Error {
    public constructor(message: any, ...params: any[]) {
        super(...params);
        this.message = message;
        /* instanbul ignore else */
        if (Error.captureStackTrace) {
            Error.captureStackTrace(this, InvalidArgumentsError);
        }
    }
}

export const InvalidArgumentError = InvalidArgumentsError;


export class SystemMessage extends Error {
    msg: INode;
    level: number;
    public constructor(msg: INode, level: number, ...params: any[]) {
        super(...params);
        this.message = msg.astext();
        this.msg = msg;
        this.level = level;
        /* instanbul ignore else */
        if (Error.captureStackTrace) {
            Error.captureStackTrace(this, SystemMessage);
        }
    }
}

export class ApplicationError extends Error {
    error: Error;
    args: any[];
    public constructor(...params: any[]) {
        super(...params);
        this.args = params;
        const [message, kwargs] = params; // eslint-disable-line no-unused-vars
        this.error = kwargs ? kwargs.error : undefined;
        /* instanbul ignore else */
        if (Error.captureStackTrace) {
            Error.captureStackTrace(this, Error);
        }
    }
}

export class DataError extends ApplicationError {
    public constructor(...params: any[]) {
        super(...params);
        /* instanbul ignore else */
        if (Error.captureStackTrace) {
            Error.captureStackTrace(this, DataError);
        }
    }
}

export class AssertError extends Error {
    public constructor(message: string, ...params: any[]) {
        super(...params);
        /* instanbul ignore else */
        if (Error.captureStackTrace) {
            Error.captureStackTrace(this, AssertError);
        }
        this.message = message;
    }
}
