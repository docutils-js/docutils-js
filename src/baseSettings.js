const default_error_encoding_error_handler = 'backslashreplace';

export default {
    tocBacklinks: 'entry',
    footnoteBacklinks: true,
    sectionNumbering: true,
    reportLevel: 2,
    haltLevel: 4,
    exitStatusLevel: 5,
    debug: true,
    traceback: null,
    inputEncodingErrorHandler: 'strict',
    outputEncoding: 'utf-8',
    outputEncodingErrorHandler: 'strict',
    errorEncodingErrorHandler: default_error_encoding_error_handler,
    errorEncoding: 'utf-8',
    languageCode: 'en',
    recordDependencies: null,
    idPrefix: '',
    autoIdPrefix: 'id',
    dumpSettings: null,
    dumpInternals: null,
    dumpTransforms: null,
    dumpPsuedoXml: null,
    exposeInternalAttribute: null,
    strictVisitor: null,
    warningStream: null,
};
