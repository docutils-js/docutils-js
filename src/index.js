export const __version__ = '0.14js';
