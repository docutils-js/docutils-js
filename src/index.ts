/* eslint-disable-next-line import/prefer-default-export */
export { parse } from './parse';
export { StringOutput, StringInput } from './io';
export { StandaloneReader } from './readers/standalone';
export { newDocument } from './newDocument';
import * as nodes from './nodes';
import Writer from './Writer';
import Transform from './Transform';
export { Reader } from './Reader';
export { Publisher } from './Publisher'
export { defaults } from '../gen/defaults';
export { pojoTranslate } from './fn/pojoTranslate';
export { htmlTranslate } from './fn/htmlTranslate';
export { RSTParser } from './parsers/restructuredtext';
export { XMLWriter } from './writers/xml'

/* eslint-disable-next-line @typescript-eslint/no-unused-vars,no-unused-vars */
const __docformat__ = 'reStructuredText';

export const __version__ = '0.14js';

export {
    nodes, Writer, Transform,
};
