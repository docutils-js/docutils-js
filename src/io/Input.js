import TransformSpec from '../TransformSpec';

class Input extends TransformSpec {
    constructor({
 source, sourcePath, encoding, errorHandler,
}) {
	super();
	this.encoding = encoding;
	this.errorHandler = errorHandler;
	this.source = source;
	this.sourcePath = sourcePath;
	if (!sourcePath) {
	    this.sourcePath = this.defaultSourcePath;
	}
	this.successfulEncoding = undefined;
    }

    /* istanbul ignore method */
    async read() {
	throw new Error('not implemented');
    }

    /* istanbul ignore method */
    decode(data) {
	return data;
    }

    /* istanbul ignore method */
    toString() {
	return `Input<${this.constructor.name}>`;
    }
}

export default Input;
