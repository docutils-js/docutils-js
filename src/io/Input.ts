import TransformSpec from '../TransformSpec';
import { ReadCallback, ReadInputCallback, InputArgs } from "../types";

abstract class Input extends TransformSpec {
    public componentType: string = "input";
    public supported: string[] = [];
    private successfulEncoding: string | undefined;
    private defaultSourcePath?: string;
    private encoding?: string;
    private errorHandler?: string;
    public sourcePath?: string;
    protected source?: any;
    public constructor(
        args: InputArgs) {
        super();
        const { source, sourcePath, encoding, errorHandler } = args;
        this.encoding = encoding;
        this.errorHandler = errorHandler;
        this.source = source;
        this.sourcePath = sourcePath;
        if (!sourcePath) {
            this.sourcePath = this.defaultSourcePath;
        }
        this.successfulEncoding = undefined;
    }

    /* istanbul ignore method */
    abstract read(cb: ReadInputCallback<string | string[] | {}>): void;

    /* istanbul ignore method */
    public decode(data: string): string {
        return data;
    }

    /* istanbul ignore method */
    public toString(): string {
        return `Input<${this.constructor.name}>`;
    }
}

export default Input;
