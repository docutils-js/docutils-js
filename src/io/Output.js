import TransformSpec  from '../TransformSpec';

class Output extends TransformSpec {
    constructor(destination, destinationPath, encoding, errorHandler) {
	super({});
	this.encoding = encoding;
	this.errorHandler = errorHandler || 'strict'
	this.destination = destination;
	this.destinationPath = destinationPath;
	if(!destinationPath) {
	    this.destinationPath = this.defaultDestinationPath;
	}
    }

    write(data) {
	process.stdout.write(data);
    }

    encode(data) {
    }

    toString() {
	return `Output<${this.constructor.name}>`;
    }

}

export default Output;
