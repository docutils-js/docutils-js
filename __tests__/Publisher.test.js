import Publisher from "../src/Publisher";
import Reader from "../src/Reader";
import Writer from "../src/Writer";
import Output from "../src/io/Output";
import Input from "../src/io/Input";
import baseSettings from "../src/baseSettings";
import { ApplicationError } from "../src/Exceptions";
import { newDocument } from "../src/utils";

const publishingSettings = baseSettings;

jest.mock("../src/Reader");
jest.mock("../src/Writer");
jest.mock("../src/io/Output");
jest.mock("../src/io/Input");

const input = [];

beforeEach(() => {
  // Clear all instances and calls to constructor and all methods:
  input.length = 0;

  Input.mockClear();
  Input.mockImplementation(({ source }) => {
    return {
      getTransforms: () => [],
      read: cb => {
        cb(undef, newDocument({}, publishSettings));
      },
      decode: data => data
    };
  });

  Reader.mockClear();
  Reader.mockImplementation((parser, parserName, args) => {
    return {
      getTransforms: () => [],
      setParser: parserName => {},
      parse: () => {
        parser.parse(/*input,document*/);
      },
      newDocument: () => newDocument({}, publishingSettings),
      read: (source, parser, settings, cb) => {
        // call this.source.rad ??
//        console.log(`in read ${cb}`);
        cb(
          undefined,
          /* expects document */ newDocument({}, publishingSettings)
        );
      }
    };
  });
  Writer.mockClear();
  Writer.mockImplementation(args => {
    return {
      getTransforms: () => [],
      write: (document, destination) => {
//        console.log(`in write ${document}`);
        return document;
      },
      assembleParts: () => {}
    };
  });

  Output.mockClear();
});

test("Instantiate publisher", () => {
  const reader = new Reader();
  const writer = new Writer();
  const source = new Input({});
  const destination = new Output();
//  console.log(Output);
//  console.log(reader);
  // destination is the correspondence to 'input'
  // which is called 'source'
//  console.log(source);
  const publisher = new Publisher({
    source,
    reader,
    writer,
    destination,
    settings: baseSettings
  });
  publisher.publish({}, (error, output) => {
      if (error) {
	  throw error;
      }
      const write = destination.write;
      expect(output.toString()).toMatchSnapshot();
  });
});

test("Instantiate publisher #2", () => {
  const reader = new Reader();
  const writer = new Writer();
  const source = new Input({});
  const destination = new Output();
  const publisher = new Publisher({ source, reader, writer, destination, settings: baseSettings });
  publisher.publish({}, (error, output) => {
      if (error) {
	  throw error;
      }
      const write = destination.write;
      expect(output.toString()).toMatchSnapshot();
  });
});
