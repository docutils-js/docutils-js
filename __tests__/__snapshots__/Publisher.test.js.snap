// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instantiate publisher #2 1`] = `"<document/>"`;

exports[`Instantiate publisher 1`] = `"<document/>"`;
